app/models/article.rb

```diff
@@ -0,0 +1,6 @@
+class Article < ApplicationRecord
+  has_many :responses
```

Comment:
It's best practice to always define your dependent action in your `has_many` associations. This defines what happens to the associated records when the parent record is destroyed. This line for instance could be written as:
```ruby
has_many :responses, dependent: :destroy
```
...which means you want to associated responses to be destroyed when the parent Article is destroyed.
Check [here](https://guides.rubyonrails.org/association_basics.html#dependent), [here](https://apidock.com/rails/ActiveRecord/Associations/ClassMethods/has_many) and [here](https://makandracards.com/makandra/32175-don-t-forget-automatically-remove-join-records-on-has_many-through-associations) for more information and other options that could be passed to  the `dependent` key.

Reply:
The same thing goes for all your `has_many`'s